### PR TITLE
Add an overview about reference counting.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

File: docs/06.REFERENCE-COUNTING.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+## Reference counting in JerryScript
+
+In JerryScript all values (jerry_value_t) are independent
```
Comment (LaszloLango) [inline docs/06.REFERENCE-COUNTING.md]: I'd use ` (code insertion) to highlight types and function names instead of * (*italic*)

**Comment by LaszloLango** [comment]
Looks good in general. Must be added to jerryscript.net and GH wiki too.

**Comment by brianjjones** [comment]
This is really useful info and helped me better understand the proper way to use jerry_create and jerry_release. Thanks for adding it.

**Comment by zherczeg** [comment]
Reworked the documentation. Now I use live/dead references to explain things.

**Comment by LaszloLango** [comment]
LGTM

File: docs/06.REFERENCE-COUNTING.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+```c
+  /* There are no 'ifs' in this snippet. */
+  jerry_release_value (jerry_set_property (..., new_prop_value));
+  jerry_release_value (new_prop_value);
+```
+
+The reference returned by a `jerry_external_handler_t` callback
+transfers the ownership of the live reference. Otherwise the
+referenced object could be freed by the garbage collector.
```
Comment (grgustaf) [inline docs/06.REFERENCE-COUNTING.md]: Thanks for this. I had deduced it but it's really nice to have it spelled out somewhere. :)

@brianjjones @jimmy-huang @jprestwo, check to make sure you understand this (sample code below).

**Pull request merged**